# IBM Telco Transfer Benchmark

Cell2Cell에서 학습한 현재 기본 모델이 IBM Telco 데이터에 직접 적용되는지 확인한 결과를 정리한 노트북이다.

- Full transfer: Cell2Cell 원본 피처 기반 모델을 IBM Telco에 투영
- Portable transfer: 범용성용 축소 피처를 IBM Telco에 투영
- IBM in-domain: IBM Telco 자체 피처로 학습한 기준선


In [1]:
from pathlib import Path
import pandas as pd

churn_rate = 0.2654
transfer = pd.DataFrame([
    ['Full transfer', 0.2657, 0.3811, 0.1321, 0.2392, 0.1702, 0.7343],
    ['Portable transfer', 0.2987, 0.5295, 0.0797, 0.0733, 0.0764, 0.7013],
    ['IBM in-domain', 0.8388, 0.7473, 0.5162, 0.7674, 0.6172, None],
], columns=['Setting', 'ROC-AUC', 'Accuracy', 'Precision', 'Recall', 'F1', 'Inverted AUC'])

print(f'IBM churn rate: {churn_rate:.4f}')
print('
Transfer and baseline comparison')
print(transfer.to_markdown(index=False))
print('
Threshold diagnostics')
print('Full transfer: best_t=0.050, best_f1=0.4194')
print('Portable transfer: best_t=0.050, best_f1=0.4194')
print('IBM in-domain: best_t=0.625, best_f1=0.6227')


IBM churn rate: 0.2654

Transfer and baseline comparison
| Setting |   ROC-AUC |   Accuracy |   Precision |   Recall |     F1 |   Inverted AUC |
|:------------------|----------:|-----------:|------------:|---------:|-------:|---------------:|
| Full transfer | 0.2657 | 0.3811 | 0.1321 | 0.2392 | 0.1702 | 0.7343 |
| Portable transfer | 0.2987 | 0.5295 | 0.0797 | 0.0733 | 0.0764 | 0.7013 |
| IBM in-domain | 0.8388 | 0.7473 | 0.5162 | 0.7674 | 0.6172 | nan |

Threshold diagnostics
Full transfer: best_t=0.050, best_f1=0.4194
Portable transfer: best_t=0.050, best_f1=0.4194
IBM in-domain: best_t=0.625, best_f1=0.6227


## 해석

- Full transfer와 Portable transfer가 둘 다 ROC-AUC 0.5 아래라서, 현재 Cell2Cell 모델은 IBM Telco에 직접 이식하기 어렵다.
- Inverted AUC가 0.70 이상으로 올라가므로, 단순 무작위가 아니라 feature 의미가 뒤집힌 상태에 가깝다.
- IBM in-domain 모델은 훨씬 강하므로, 다음 단계는 IBM Telco에 맞는 공통 피처를 다시 설계하거나 IBM 데이터로 재학습하는 것이다.
